### DISCUSSION TITLE
specifying virtual atoms

### DISCUSSION THREAD

**Discussion by rfhari** [brucefan1983/GPUMD#915] [General]
Hello, I am working on implementing the Virtual Crystal Approximation (VCA) in molecular dynamics simulations for alloys, such as the Al(x)Ga(1-x)N system. I've interpolated the second- and third-order force constants (FC2 and FC3) based on the Al composition to use as inputs. My plan was to utilize the [Force Constant Potential (FCP)](https://gpumd.org/potentials/fcp.html) to run a short MD simulation on virtual crystal and compare the forces obtained from the interpolated FC2 and FC3 with those from a fully disordered system.

However, I encountered an issue: FCP seem to require a list of atoms involved in the force constant calculations to be specified in the [driver file](https://gpumd.org/potentials/fcp.html#driver-file). In the VCA, these atoms are "virtual" atoms, with linearly interpolated mass between Al and Ga. Could you please advise if there’s a way to incorporate such virtual atoms by modifying their mass in the [equilibrium position file](https://gpumd.org/potentials/fcp.html#equilibrium-position-file)? 

Thank you!

**Comment by brucefan1983** [comment]
I am not familiar with VCA. The equilibrium position file does not have information on mass. Mass is set in the `model.xyz` file via the `properties=mass:R:1` tag. Can this solve your problem?

`model.xyz` file format: 

https://gpumd.org/gpumd/input_files/model_xyz.html#model-xyz
